jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -1029,7 +1028,10 @@ public static ByteBuffer toBuffer(Resource resource, boolean direct) throws IOEx
             BufferUtil.readFrom(resource.getPath(), buffer);
         else
         {
-            try (InputStream is = resource.getInputStream();)
+            // TODO what does it mean to create a buffer from a resource that is not backed by a Path?
+            //  For instance, passing an EmptyResource to this method allocates and returns an empty buffer
+            //  while we could test for if (path == null) return EMPTY_BUFFER.
```

Comment:
I think this is the wrong question.  Can all resources with a path be `readFrom`?   What if it is a directory?

Reply:
A directory cannot be read in the traditional sense.

Depending on the OS, a read from a directory might return directory metadata.

Reply:
So either:
 + we throw ISE if you try to read a directory
 + we return an empty InputStream
 + we return an InputStream over the HTML listing of the directory.

I like ISE for now.

Reply:
I like throwing ISE